01-algorithms/02-algorithms-divide-conquer.md

```diff
@@ -45,205 +27,15 @@ QuickSort is an algorithm which takes a divide-and-conquer approach to sorting a
-
-For example, if with each iteration the pivot is the smallest remaining element in an array, rather than splitting the array into approximately _n/2_ sized pieces, instead we have one subarray of size _n-1_.
-
-<br />
-
-<details>
-
-<summary>The example implementation provided here uses the approach of taking the final element as the pivot, then gradually growing the smaller list from the left side of the non-pivot elements.</summary>
-
-```python
-def quicksort(array, low=0, high=None):
-    if high is None:
-        high = len(array)
-
-    # if the array has 0 or 1 elements, it's already sorted
-    if high - low <= 1:
-        return
-
-    # partition the current list to find where one element goes
-    partition_pos = partition(array, low, high)
-
-    # sort the left and right sides
-    quicksort(array, low, partition_pos)
-    quicksort(array, partition_pos + 1, high)
-
-def partition(array, low, high):
-    # take the last item as the pivot
-    last = high - 1
-    pivot = array[last]
-
-    # assume pivot will end up at the start
-    # conceptually, this position marks the end of the smaller list
-    p_index = low
-
-    # iterate over the values not including the pivot
-    i = low
-    while i < last:
-        # if the current value should be to the left of the pivot, swap
-        # this value to the potential pivot location and advance that
-        # location. conceptually this adds the value to the end of the
-        # smaller list and tracks the new end of this list
-        if array[i] < pivot:
-            temp = array[i]
-            array[i] = array[p_index]
-            array[p_index] = temp
-            p_index += 1
-        i += 1
-
-    # swap the pivot value to the end of the smaller list. after this
-    # swap, we know that all values to the left of the pivot are smaller,
-    # and all values to the right of the pivot are greater or equal
-    temp = array[i]
-    array[i] = array[p_index]
-    array[p_index] = temp
-
-    return p_index
-```
-
-</details>
-
-### !callout-info
-
-## Partitioning Is the Key
-
-There are a variety of strategies for selecting the pivot, and for how to rearrange the elements around the pivot. We will not go into further detail about them here, but follow your curiosity!
-
-### !end-callout
-
-## Example: Merge Sort
+### Example: Merge Sort
 
 Merge sort is another divide-and-conquer algorithm. It involves the following three stages:
 
 1. _Divide_ the array into two subarrays at each step until each subarray is of size one.
 2. _Sort_ each subarray with the merge sort algorithm. (An array of size one is trivially sorted.)
 3. _Merge_ the subarrays into one array by combining two subarrays into one at each step.
 
-This process is usually done by keeping track of three indices in the array: _starting index_, _ending index_ and _midway index_ as shown in the image below.
```

Comment:
Same here, I actually really like the photo we provide

Reply:
Added this back. Thank you for the suggestion!